python/ray/tests/test_environ.py

```diff
@@ -31,6 +32,33 @@ def get_os_environ(self):
     assert len(actor_os_environ) > 0
 
 
+def test_update_envs():
+    env_vars = {
+        "PATH": "/test/lib/path:${PATH}",
+        "LD_LIBRARY_PATH": "/test/path1:${LD_LIBRARY_PATH}:./test/path2",
+        "DYLD_LIBRARY_PATH": "${DYLD_LIBRARY_PATH}:/test/path",
+        "LD_PRELOAD": "",
+    }
+    old_path = os.environ["PATH"]
+    os.environ["LD_LIBRARY_PATH"] = "./"
+    os.environ["DYLD_LIBRARY_PATH"] = "/lib64"
+    os.environ["LD_PRELOAD"] = "/lib:/usr/local/lib"
+    update_envs(env_vars)
+    assert os.environ["PATH"] == "/test/lib/path:" + old_path
+    assert os.environ["LD_LIBRARY_PATH"] == "/test/path1:./:./test/path2"
+    assert os.environ["DYLD_LIBRARY_PATH"] == "/lib64:/test/path"
+    assert os.environ["LD_PRELOAD"] == env_vars["LD_PRELOAD"]
+
+    # Test the empty string scenario
+    os.environ["LD_LIBRARY_PATH"] = ""
+    del os.environ["DYLD_LIBRARY_PATH"]
+    del os.environ["LD_PRELOAD"]
+    update_envs(env_vars)
+    assert os.environ["LD_LIBRARY_PATH"] == "/test/path1::./test/path2"
+    assert os.environ["DYLD_LIBRARY_PATH"] == env_vars["DYLD_LIBRARY_PATH"]
+    assert os.environ["LD_PRELOAD"] == env_vars["LD_PRELOAD"]
```

Comment:
The test should use `patch` so that it doesn't affect global state. example https://github.com/ray-project/ray/blob/0dafee965c96c51dd118bb56dab448cbfdeb2bb1/python/ray/tests/test_ray_debugger.py#L343